Re-factor translations of Auhority roles

eronisko opened the pull request:
- Authority roles (e.g. painter, sculptor, ...) are now stored as an array with values directly taken from CEDVU
- Translations are handled using lang files
- SK translations have support for 'gendered' forms
- Fallback for missing Czech translations (for roles) is now set to Slovak
- Translations are indexed translated in male form (this is not a change)

⚠️ This change will drop all roles data from the database. Authorities will need to be re-harvested.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [ ] I have updated the [CHANGELOG](../CHANGELOG.md)
- [ ] I have requested at least 1 reviewer for this PR
- [ ] I have made corresponding changes to the documentation

rastislav-chynoransky commented on resources/views/frontend/authors/index.blade.php:
```diff
@@ -76,9 +76,11 @@ class="icon-cross"></i></a>
                             </div>
                             <div>
-                                @foreach ($author->roles as $i => $role)
+                                @foreach ($author->translatedRoles as $i => $role)
```
```suggestion
                                @foreach ($author->translatedRoles as $role)
```

rastislav-chynoransky commented on resources/views/frontend/authors/index.blade.php:
```diff
@@ -76,9 +76,11 @@ class="icon-cross"></i></a>
-                                        href="{!! route('frontend.author.index', ['role' => $role]) !!}"><strong>{!! trans_choice($role, $author->sex) !!}</strong></a>{!! $i + 1 < count($author->roles) ? ', ' : '' !!}
+                                        href="{{ route('frontend.author.index', ['role' => $role->indexed]) }}">
+                                        <strong>{{ $role->formatted }}</strong>
+                                    </a>{!! $i + 1 < count($author->translatedRoles) ? ', ' : '' !!}
```
```suggestion
                                    </a>{{ !$loop->last ? ', ' : '' }}
```

rastislav-chynoransky commented on config/authorityRoles.php:
```diff
@@ -1,33 +1,33 @@
-    "dielňa autora/workshop of" =>                           "authority.role.workshop",
-    "autor konceptu/concept by" =>                           "authority.role.concept",
-    "autor fotografie/photographer" =>                       "authority.role.photograph",
+    "autor/author" =>                                        "item.authority_roles.author",
+    "autor predlohy/after" =>                                "item.authority_roles.after",
+    "ateliér/atelier" =>                                     "item.authority_roles.atelier",
+    "okruh autora/circle of" =>                              "item.authority_roles.circle",
+    "kopista/copyist of" =>                                  "item.authority_roles.copyist",
+    "autor návrhu/draft by" =>                               "item.authority_roles.draft",
+    "kresliar/draftsman" =>                                  "item.authority_roles.drawer",
+    "rytec/engraver" =>                                      "item.authority_roles.engraver",
+    "napodobňovateľ/epigone of" =>                           "item.authority_roles.epigone",
+    "nasledovník/follower of" =>                             "item.authority_roles.follower",
+    "pôvodné určenie/formerly attributed to" =>              "item.authority_roles.former",
+    "grafik/graphic artist" =>                               "item.authority_roles.graphic",
+    "autor úpravy záznamu/modified by" =>                    "item.authority_roles.modifier",
+    "štúdio/office of" =>                                    "item.authority_roles.office",
+    "autor origináu/original by" =>                          "item.authority_roles.original",
+    "tlačiar/printer" =>                                     "item.authority_roles.printer",
+    "pravdepodobne/probably by" =>                           "item.authority_roles.probably",
+    "pravdepodobný autor predlohy/probably after" =>         "item.authority_roles.probablyAfter",
+    "pravdepodobne okruh autora/probably circle of" =>       "item.authority_roles.probablyCircle",
+    "pravdepodobne kresliar/probable draftsman" =>           "item.authority_roles.probablyDrawer",
+    "pravdepodobne rytec/probable engraver" =>               "item.authority_roles.probablyEngraver",
+    "pravdepodobne tlačiar/probable printer" =>              "item.authority_roles.probablyPrinter",
+    "pravdepodobne dielňa autora/probably workshop of" =>    "item.authority_roles.probablyWorkshop",
+    "výrobca/producer" =>                                    "item.authority_roles.producer",
+    "vydavateľ/publisher" =>                                 "item.authority_roles.publisher",
+    "reštaurátor/restorer" =>                                "item.authority_roles.restorer",
+    "dielňa autora/workshop of" =>                           "item.authority_roles.workshop",
+    "autor konceptu/concept by" =>                           "item.authority_roles.concept",
+    "autor fotografie/photographer" =>                       "item.authority_roles.photograph",
```
I find this mapping unnecessary as we could use the config keys as translation keys.

eronisko replied:
I agree but since this PR deals with Authority roles (and Item->authorityRoles are just moved around), I'd like to fix this at a different time

rastislav-chynoransky reviewed (changes requested):
Thanks, this is huge improvement. 🙏
I'm putting some suggestions here. However, one general is that this PR mixes two things: item authority roles and authority roles. I think it would be better to also split it up into two PR's.

eronisko commented on app/Authority.php:
```diff
@@ -215,6 +211,25 @@ public function getFormatedNamesAttribute()
+        return collect($this->roles)
+            ->map(function ($role) {
+                // Use SK as fallback locale
+                $locale = Lang::hasForLocale("authority.roles.$role") ? Lang::getLocale() : 'sk';
```
If SK fallback is not suitable then I'd rather show nothing (which is the current state anyway)

rastislav-chynoransky replied:
You're right, it's out of scope of this PR, but it would be nice to create a new task for it till we still remember it's missing.

eronisko commented on app/Authority.php:
```diff
@@ -215,6 +211,25 @@ public function getFormatedNamesAttribute()
+                }
+
+                return (object) [
+                    'indexed' => Lang::choice("authority.roles.$role", 'male', [], $locale),
+                    'formatted' => Lang::choice("authority.roles.$role", $this->sex, [], $locale),
```
It is, but this here is just a carry-over from https://github.com/SlovakNationalGallery/webumenia.sk/pull/963/files#diff-9085f5c4cf5c557542e242ccc3eb661124d60518cfb7f6c7f989791c404519eaL40

I don't have a better solution for now and this does a good job with how it works and fallbacks. I considered using `authority.roles.painter.female` etc but then it gets tricky between EN and SK/CZ.

eronisko commented:
Thank you @rastislav-chynoransky for a thorough review! I've addressed most of your comments apart from this:
> However, one general is that this PR mixes two things: item authority roles and authority roles. I think it would be better to also split it up into two PR's.

This PR does touch item authority roles but it's mostly to move their translation strings in the `item` domain. So a painter is translated as `authority.roles.painter` but an item's restorer is translated under `item.authority_roles.restorer`. Does that make sense? Or is there a mix-up somewhere where I missed it?

eronisko commented on app/Forms/Types/ItemAuthoritiesType.php:
```diff
@@ -36,7 +36,7 @@ public function buildView(FormView $view, FormInterface $form, array $options)
             return  [
                 "key" => $role,
-                "value" => trans('authority.role.' . config('authorityRoles')[$role])
+                "value" => trans('item.authority_roles.' . config('authorityRoles')[$role])
```
I think I forgot to test this -- thanks

rastislav-chynoransky commented:
> Thank you @rastislav-chynoransky for a thorough review! I've addressed most of your comments apart from this:
> 
> > However, one general is that this PR mixes two things: item authority roles and authority roles. I think it would be better to also split it up into two PR's.
> 
> This PR does touch item authority roles but it's mostly to move their translation strings in the `item` domain. So a painter is translated as `authority.roles.painter` but an item's restorer is translated under `item.authority_roles.restorer`. Does that make sense? Or is there a mix-up somewhere where I missed it?

Ah, sure, right, it's just resolving the conflict. Good to go!

rastislav-chynoransky reviewed (approved):
🚀
